bundles/org.openhab.binding.fsinternetradio/src/main/java/org/openhab/binding/fsinternetradio/internal/handler/FSInternetRadioHandler.java

```diff
@@ -45,21 +47,27 @@
  * @author Mihaela Memova - removed the unused boolean parameter, changed the check for the PIN
  * @author Svilen Valkanov - changed handler initialization
  */
+@NonNullByDefault
 public class FSInternetRadioHandler extends BaseThingHandler {
 
     private final Logger logger = LoggerFactory.getLogger(FSInternetRadioHandler.class);
 
-    FrontierSiliconRadio radio;
+    public @Nullable FrontierSiliconRadio radio;
     private final HttpClient httpClient;
 
     /** Job that runs {@link #updateRunnable}. */
-    private ScheduledFuture<?> updateJob;
+    private @Nullable ScheduledFuture<?> updateJob;
 
     /** Runnable for job {@link #updateJob} for periodic refresh. */
     private final Runnable updateRunnable = new Runnable() {
         @Override
         public void run() {
-            if (!radio.isLoggedIn()) {
+            FrontierSiliconRadio localRadio = radio;
```

Comment:
Can we do it this way instead?
```suggestion
            FrontierSiliconRadio radio = this.radio;
```

this would be consistent with your `updateJob` change in the same class (further down), and also wouldn't wouldn’t evoke associations with [local radio](https://tunein.com/radio/local/). 😉 (btw, thanks for this example that I can add to my `String localHost = host` allergy 😁)

Reply:
You could still do that though with a small tweak:
```java
FrontierSiliconRadio radio = FSInternetRadioHandler.this.radio;
```